### PR TITLE
feat: add support for creating velero schedules (#350)

### PR DESCRIPTION
**Pull request by kosmoz**

### CODE REVIEW

File: operator/src/main/kotlin/eu/glasskube/operator/apps/common/backup/BackupSpec.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package eu.glasskube.operator.apps.common.backup
+
+import eu.glasskube.operator.apps.common.cloudstorage.CloudStorageSpec
+import io.fabric8.generator.annotation.Nullable
+import io.fabric8.generator.annotation.Required
+import io.fabric8.kubernetes.api.model.SecretKeySelector
+
+data class BackupSpec(
+    val schedule: String = "0 0 4 * * *",
```
Comment (pmig) [inline operator/src/main/kotlin/eu/glasskube/operator/apps/common/backup/BackupSpec.kt:9]: Velero schedule only allows 5 field cron annotation

```
status:
  phase: FailedValidation
  validationErrors:
    - 'invalid schedule: Expected exactly 5 fields, found 6: 0 0 4 * * *
```

Comment (pmig) [reply operator/src/main/kotlin/eu/glasskube/operator/apps/common/backup/BackupSpec.kt:9]: It also seems that the schedule doesn't get updated if changed and can't be changed via the CRD

**Review by pmig — CHANGES_REQUESTED** [review]
The velero s3 path looks like: `restic/matomo/config` although the backup name is `matomo.matomo-matomo` maybe the dot will not be added to the backup path (which is not ideal)

beside that it works fine :-)

**Comment by kosmoz** [comment]
> The velero s3 path looks like: `restic/matomo/config` although the backup name is `matomo.matomo-matomo` maybe the dot will not be added to the backup path (which is not ideal)

So, I did some digging and it looks like this is a limitation of Veleros "File System Backup" component. When instructed to create a PodVolumeBackup during a backup, the `BackupRepoReconciler` ensures that a `BackupRepository` exists for the `Pod`s namespace and the `BackupStorageLocation` of the `Backup`, using only the `Pod`s namespace as part of the `resticIdentifier` (regardless of the uploaderType used). 

I think the only workaround for this might be to create a `BackupRepository` before any `Backup`s are scheduled, but I'm not sure if that would work.

File: operator/src/main/kotlin/eu/glasskube/operator/apps/common/backup/BackupSpec.kt (hunk 1 above)
Comment (kosmoz) [reply operator/src/main/kotlin/eu/glasskube/operator/apps/common/backup/BackupSpec.kt]: Fixed

**Comment by kosmoz** [comment]
added Cloud Storage Backup CronJob for Gitlab and updated Cloud Storage Backup image tag to v0.2.1

**Review by pmig — APPROVED** [review]

**Pull request merged**